src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -206,9 +208,9 @@ private void refreshSecretFromCloud(String arn, String versionStage) {
         String versionLabel = Utils.isEmpty(versionStage) ? LATEST_LABEL : versionStage;
         List<SecretConfiguration> configurations = getSecretConfiguration();
         boolean isSecretLabelConfigured = configurations.stream().anyMatch(
-                (secret) -> secret.getArn().equalsIgnoreCase(arn) && secret.getLabels().contains(versionStage));
+                (secret) -> secret.getArn().contains(arn) && secret.getLabels().contains(versionStage));
```

Comment:
From our previous discussion, the suggestion is to separate this check for secret label configuration out to a different method and have the test coverage with secret arn, partial arn and a friendly name.

Reply:
I can update this as part of a different PR. My initial understanding is that label validation will take a bit more work and should not be a release candidate right now.